packages/ironfish-native-module/rust_lib/src/lib.rs

```diff
@@ -159,49 +159,118 @@ pub fn read_partial_file(path: String, offset: u32, length: u32) -> Vec<u8> {
 
 #[uniffi::export(async_runtime = "tokio")]
 pub async fn decrypt_notes_for_owner(
-    note_encrypted: Vec<String>,
+    note_encrypteds: Vec<String>,
     incoming_hex_key: String,
 ) -> Result<Vec<DecryptedNote>, EnumError> {
     let incoming_view_key = ironfish::IncomingViewKey::from_hex(&incoming_hex_key)
         .map_err(|e| EnumError::Error { msg: e.to_string() })?;
 
-    let idxes = note_encrypted
+    let idxes = note_encrypteds
         .par_iter()
         .enumerate()
         .filter_map(|(i, output)| {
             let bytes = const_hex::decode(output);
             if bytes.is_err() {
-                println!("error converting hex to bytes");
+                eprintln!("error converting hex to bytes");
                 return None;
             }
 
             let note = ironfish::MerkleNote::read(bytes.unwrap().as_slice());
             if note.is_err() {
-                println!("error reading bytes");
+                eprintln!("error reading bytes");
                 return None;
             }
 
             let dec: Result<ironfish::Note, ironfish::errors::IronfishError> =
                 note.unwrap().decrypt_note_for_owner(&incoming_view_key);
-            if dec.is_ok() {
-                let mut vec = vec![];
-                if dec
-                    .unwrap()
-                    .write(&mut vec)
-                    .map_err(|e| EnumError::Error { msg: e.to_string() })
-                    .is_err()
-                {
-                    println!("error writing bytes");
-                    return None;
-                }
-
-                return Some(DecryptedNote {
-                    index: i as u32,
-                    note: const_hex::encode(&vec),
-                });
+
+            if dec.is_err() {
+                return None;
+            }
+
+            let mut vec = vec![];
+            if dec
+                .unwrap()
+                .write(&mut vec)
+                .map_err(|e| EnumError::Error { msg: e.to_string() })
+                .is_err()
+            {
+                eprintln!("error writing bytes");
+                return None;
             }
-            None
+
+            Some(DecryptedNote {
+                index: i as u32,
+                note: const_hex::encode(&vec),
+            })
         });
 
     Ok(idxes.collect())
 }
+
+#[uniffi::export(async_runtime = "tokio")]
+pub async fn decrypt_notes_for_spender(
+    note_encrypteds: Vec<String>,
```

Comment:
More curious than anything, is naming things like this a convention? I feel like `encrypted_notes` would read more naturally

Reply:
Not really, I just named it that since that's what the `ironfish-rust-nodejs` code calls it: https://github.com/iron-fish/ironfish/blob/master/ironfish-rust-nodejs/src/structs/note_encrypted.rs I agree that it sounds weird

I think `NoteEncrypted` was originally named that way so that in an alphabetically-sorted list, `Note` and `NoteEncrypted` would appear next to each other.